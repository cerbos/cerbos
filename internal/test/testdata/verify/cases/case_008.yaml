description: Duplicate test (#1410)
