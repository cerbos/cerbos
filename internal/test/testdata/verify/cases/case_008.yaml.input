-- testdata/principals.yaml --
---
principals:
  john:
    id: john
    policyVersion: '20210210'
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design
  alicia:
    id: alicia
    policyVersion: '20210210'
    roles:
      - employee
      - manager
    attr:
      department: marketing
      geography: GB
      managed_geographies: GB
      ip_address: 10.20.1.2
      team: design

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design

-- suite_test.yaml --
---
name: TestSuite
description: Tests for verifying something
tests:
  - name: John and his leave request
    input:
      principals:
        - john
        - alicia
      resources:
        - john_leave_request
      actions:
        - view:public
        - approve
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW
          approve: EFFECT_DENY

  - name: Alicia approves John leave request
    input:
      principals:
        - alicia
      resources:
        - john_leave_request
      actions:
        - approve
    expected:
      - principal: alicia
        resource: john_leave_request
        actions:
          approve: EFFECT_ALLOW
