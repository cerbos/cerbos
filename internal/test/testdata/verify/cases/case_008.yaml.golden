{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_FAILED",
        "testsCount": 5,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 3
          },
          {
            "result": "RESULT_FAILED",
            "count": 2
          }
        ]
      },
      "testCases": [
        {
          "name": "John and his leave request",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED"
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED"
                      }
                    }
                  ]
                }
              ]
            },
            {
              "name": "alicia",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_FAILED",
                        "failure": {
                          "expected": "EFFECT_DENY",
                          "actual": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED"
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "Alicia approves John leave request",
          "principals": [
            {
              "name": "alicia",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_FAILED",
                        "failure": {
                          "expected": "EFFECT_ALLOW",
                          "actual": "EFFECT_DENY"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_FAILED",
    "testsCount": 5,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 3
      },
      {
        "result": "RESULT_FAILED",
        "count": 2
      }
    ]
  }
}
